src/ExecClient.cs

```diff
@@ -0,0 +1,124 @@
+using System;
+using System.Buffers;
+using System.Net.WebSockets;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace k8s
+{
+    public class ExecClient : IDisposable
+    {
+        private const int StdInStreamIndex = 0;
+        private const int StdOutStreamIndex = 1;
+        private const int StdErrStreamIndex = 2;
+        private const int ErrorChannelIndex = 3;
+        private const int ResizeChannelIndex = 4;
+
+        private readonly WebSocket socket;
+        private readonly CancellationTokenSource cts;
+        private Task runLoop;
+
+        public ExecClient(WebSocket socket)
+        {
+            this.socket = socket ?? throw new ArgumentNullException(nameof(socket));
+            this.cts = new CancellationTokenSource();
+        }
+
+        public event EventHandler ConnectionClosed;
+        public event EventHandler<string> StandardOutputReceived;
+        public event EventHandler<string> StandardErrorReceived;
+
+        public bool Running
+        {
+            get { return this.runLoop != null && this.runLoop.Status == TaskStatus.Running; }
+        }
+
+        public void Start()
+        {
+            if (this.runLoop == null)
+            {
+                this.runLoop = this.RunLoop(this.cts.Token);
+            }
+        }
+
+        public void Stop(CancellationToken cancellationToken = default(CancellationToken))
+        {
+            if (this.runLoop != null)
+            {
+                this.cts.Cancel();
+                this.runLoop.Wait(cancellationToken);
+            }
+        }
+
+        public void Dispose()
+        {
+            this.Stop();
+        }
+
+        public async Task Write(string command, CancellationToken cancellationToken = default(CancellationToken))
+        {
+            // Send:
+            //  - [byte: stream index = 0 for stdin]
+            //  - [byte[]: command as UTF8 value]
+            var commandLength = Encoding.UTF8.GetByteCount(command);
+            var buffer = ArrayPool<byte>.Shared.Rent(commandLength + 1);
+
+            try
+            {
+                // The first byte represent the stream index. For stdin this is 0
+                buffer[0] = 0;
```

Comment:
I think it would be better to hide channel in the `websocket` class which can be reuse in attach. but I think it is OK to impl them in next PR.